File: app/Models/Space.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Traits\Auditable;

class Space extends Model
{
    use Auditable;

    protected $fillable = ['name', 'description', 'auxiliar_id', 'laboratory_id'];

    public function auxiliar()
    {
        return $this->belongsTo(User::class, 'auxiliar_id');
    }

    public function laboratory()
    {
        return $this->belongsTo(Laboratory::class);
    }

    public function labReservations()
    {
        return $this->hasMany(LabReservation::class);
    }
}

File: mobile/lib/models/space.dart
import 'user.dart';

class Space {
  final int id;
  final String name;
  final String? description;
  final AppUser? auxiliar;

  Space({required this.id, required this.name, this.description, this.auxiliar});

  factory Space.fromJson(Map<String, dynamic> json) {
    return Space(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      auxiliar: json['auxiliar'] != null ? AppUser.fromJson(json['auxiliar'] as Map<String, dynamic>) : null,
    );
  }
}
